onadata/apps/logger/models/instance.py

```diff
@@ -125,13 +126,19 @@ def update_user_submissions_counter(sender, instance, created, **kwargs):
     )
     today = date.today()
     first_day_of_month = today.replace(day=1)
-    queryset = SubmissionCounter.objects.filter(
-        user_id=user_id, timestamp=first_day_of_month
-    )
-    if not queryset.exists():
-        SubmissionCounter.objects.create(user_id=user_id)
+    try:
+        queryset = SubmissionCounter.objects.get(
+            user_id=user_id, timestamp=first_day_of_month
+        )
+    except ObjectDoesNotExist:
+        SubmissionCounter.objects.create(user_id=user_id, timestamp=first_day_of_month)
+        queryset = SubmissionCounter.objects.get(
+            user_id=user_id,
+            timestamp=first_day_of_month,
+        )
 
-    queryset.update(count=F('count') + 1)
+    queryset.count = queryset.count + 1
+    queryset.save()
```

Comment:
I'm afraid we have. to keep `update()` with the `F` function to avoid race conditions. 
If two queries arrive at the same time, the `.save()` of the second query could overwrite the result of `count+1`.
For example: 

```python
# count = 1

#  query 1 comes in 
queryset.count = queryset.count + 1  # fetch result from DB 
# query 2 comes in. 
queryset.count = queryset.count + 1  # fetch result from DB , queryset has not been saved yet, queryset.count still equals 1
# query 2 is saved
queryset.save() # count in DB equals 2
# query 1 is saved
queryset.save() # count in DB still equals 2 
```

`queryset.update(count=F('count') + 1)` avoids the issue because the calculation is done on the database side.
I think we need to modify the commands to trigger the signal (or call the method directly)